src/components/organisms/BannerAdmin/BannerAdmin.tsx

```diff
@@ -1,89 +1,190 @@
-import React, { useState, useCallback, useRef } from "react";
+import React, { useCallback } from "react";
+import { useForm } from "react-hook-form";
+import classNames from "classnames";
 
 import { makeUpdateBanner } from "api/bannerAdmin";
 
 import { AnyVenue } from "types/venues";
+import { Banner } from "types/banner";
+
+import { useShowHide } from "hooks/useShowHide";
+
+import { Button } from "components/atoms/Button";
+import { Checkbox } from "components/atoms/Checkbox";
+import { InputField } from "components/atoms/InputField";
+import { ConfirmationModal } from "components/atoms/ConfirmationModal/ConfirmationModal";
+
+import "./BannerAdmin.scss";
+import { useAsyncFn } from "react-use";
 
 interface BannerAdminProps {
   venueId?: string;
   venue: AnyVenue;
+  onClose?: () => void;
 }
 
 // @debt This component is almost exactly the same as IframeAdmin, we should refactor them both to use the same generic base component
 //   BannerAdmin is the 'canonical example' to follow when we do this
-export const BannerAdmin: React.FC<BannerAdminProps> = ({ venueId, venue }) => {
-  const existingBannerMessage = venue?.bannerMessage ?? "";
-
-  const textareaFieldRef = useRef<HTMLTextAreaElement>(null);
-  const [error, setError] = useState<string | null>();
+export const BannerAdmin: React.FC<BannerAdminProps> = ({
+  venueId,
+  venue,
+  onClose,
+}) => {
+  const { register, handleSubmit, errors, reset, watch } = useForm<Banner>({
+    mode: "onChange",
+    reValidateMode: "onChange",
+  });
+  const isUrlButtonActive = watch(
+    "isActionButton",
+    venue?.banner?.isActionButton
+  );
 
-  const handleInputChange = useCallback(() => {
-    setError(null);
-  }, []);
+  const {
+    isShown: isShowBannerChangeModal,
+    show: showBannerChangeModal,
+    hide: hideBannerChangeModal,
+  } = useShowHide();
 
-  const updateBannerInFirestore = useCallback(
-    (msg?: string) => {
+  const [{ loading: isUpdatingBanner }, saveBanner] = useAsyncFn(
+    async (banner?: Banner) => {
       if (!venueId) return;
 
-      makeUpdateBanner(venueId, (errorMsg) => setError(errorMsg))(msg);
+      await makeUpdateBanner({ venueId, banner });
+      onClose && onClose();
```

Comment:
These can be shortened to `onClose?.()` However it's up to @0xdevalias and @mike-lvov to tell if we prefer guards or optional parameters. Consider this is a comment not a change request.

Reply:
thanks, noted